modules/core/src/main/java/org/apache/ignite/internal/processors/platform/utils/PlatformUtils.java

```diff
@@ -989,7 +990,9 @@ private static boolean knownMap(Object o) {
      * @return Result.
      */
     public static Object[] unwrapBinariesInArray(Object[] arr) {
-        Object[] res = new Object[arr.length];
+        Class<?> compType = arr.getClass().getComponentType();
```

Comment:
What is the use case when `compType` is not `Object`, is it covered with some test? I thought we are unwrapping binary objects here, so `arr` is either `Object[]`, or maybe `BinaryObject[]`.

In my understanding, `res[i] = unwrapBinary(arr[i])` line below can return an instance of any class, so `res` must be an `Object[]`.

Reply:
`unwrapBinary` keeps the original object if it's not a `BinaryObject`.
This method breaks it because converts any typed array into `Object[]`.

These changes fix it.

Reply:
Understood, thanks. Which test covers this scenario?

Reply:
.Net test `ServicesTypeAutoResolveTest#TestCallJavaService` will not pass if changes reverted.

Call of the `testOverload` method goes over change.

Reply:
I think we can change the logic to the following:
```
if (arr.getClass().getComponentType() != Object.class)
    return arr;
```

Arrays like this do not require further unwrapping.

Reply:
What about an array of maps `java.util.collections.Map[]` or something similar(`Collection[]`)?

Reply:
Can you add a test to demonstrate?

Reply:
Looks like you are right.

I've added some tests and the proposed clause.
Thanks!